hikari/impl/shard.py

```diff
@@ -368,11 +371,16 @@ async def close(self) -> None:
                 self._logger.info("received request to shut down shard")
             else:
                 self._logger.debug("shard marked as closed when it was not running")
-            self._request_close_event.set()
 
             if self._ws is not None:
-                self._logger.warning("gateway client closed, will not attempt to restart")
-                await self._close_ws(errors.ShardCloseCode.NORMAL_CLOSURE, "client shut down")
+                self._closing = True
+                # If anything interrupts this task, aiohttp will just refuse to send the close frame
+                # and we will never know anything about it.
+                await asyncio.shield(self._close_ws(errors.ShardCloseCode.GOING_AWAY, "client shut down"))
+                self._logger.info("gateway client closed, will not attempt to restart")
+
+            self._request_close_event.set()
+            self._ratelimiter.close()
```

Comment:
don't we want to close the ratelimiter before closing the ws so it doesn't try to send of a request while the ws connection is closed?

Reply:
Fixed.